### PR TITLE
feat(person-select): search for azureId with PersonInfo task and impr…

### PR DESCRIPTION
**Pull request by eikeland**
Support for searching for azureId by adding the PersonInfoTask interface and component states for `upn` and `azureId`.

Improves searching with the `selectedPerson` property.

### CODE REVIEW

File: packages/person/src/components/select/element.ts [hunk 1]
```diff
@@ -172,64 +181,78 @@ export class PersonSelectElement
    */
   protected renderList(): HTMLTemplateResult {
     if (!this.controllers.element.isOpen) {
+      if (this.selectedPerson && this.tasks.info.status === 2) {
```
Comment (asbjornhaland) [inline packages/person/src/components/select/element.ts:184]: Can you use the `TaskStatus.COMPLETE` instead? I don't like magic numbers.

Comment (eikeland) [reply packages/person/src/components/select/element.ts]: Done

File: packages/person/src/components/select/element.ts [hunk 2]
```diff
@@ -172,64 +182,78 @@ export class PersonSelectElement
...
-          `;
-        },
-        pending: () =>
-          html`<fwc-list-item disabled=${true}><fwc-dots-progress size="small" color="primary" /></fwc-list-item>`,
-        /* Error from resolvers searchQuery Promise */
-        error: (e: unknown) =>
-          html`<fwc-list-item disabled=${true} class="item-error">
-            <span class="item-text"><span class="item-title">${e}</span></span>
-          </fwc-list-item>`,
+    // ${(this.selectedPerson?.azureId || this.selectedPerson?.upn) && Object.keys(this.selectedPerson).length === 1
```
Comment (asbjornhaland) [inline packages/person/src/components/select/element.ts]: please remove dead code.

File: packages/person/src/components/select/element.ts [hunk 3]
```diff
@@ -296,7 +320,9 @@ export class PersonSelectElement
             icon=${this.leadingIcon}
             dense=${ifDefined(dense)}
             placeholder=${this.placeholder}
-            @focus=${() => (this.controllers.element.isOpen = true)}
+            @focus=${() => {
```
Comment (asbjornhaland) [inline packages/person/src/components/select/element.ts:322]: Should there be a counterpart `@blur` to set `isOpen` to `false`?

**Review by asbjornhaland — CHANGES_REQUESTED** [review]

**Review by asbjornhaland — APPROVED** [review]

**Pull request merged**